integration-tests/tests/utils/aws/s3GetAvailableTestDate.ts

```diff
@@ -0,0 +1,50 @@
+import { ListObjectsCommand } from '@aws-sdk/client-s3'
+import { getEnv } from '../helpers'
+import { s3Client } from './s3Client'
+
+export const getAvailableTestDate = async () => {
+  return findAvailableS3Locations(generateRandomDateAndPrefix())
+}
+
+const findAvailableS3Locations = async (checkDate: MatchedDateAndPrefix) => {
+  const buckets = [getEnv('ANALYSIS_BUCKET_NAME'), getEnv('AUDIT_BUCKET_NAME')]
+
+  const locationAvailability = await Promise.all(
+    buckets.map(async (bucket) => {
+      const input = {
+        Bucket: bucket,
+        Prefix: checkDate.prefix
+      }
+      const command = new ListObjectsCommand(input)
+      const response = await s3Client.send(command)
+
+      return !response.Contents || response.Contents.length === 0 ? true : false
+    })
+  )
+
+  if (locationAvailability.includes(false)) {
+    await findAvailableS3Locations(generateRandomDateAndPrefix())
+  }
+
+  return checkDate
+}
+
+const generateRandomDateAndPrefix = (): MatchedDateAndPrefix => {
+  const date = new Date(Math.random() * Date.now())
```

Comment:
What would happen if we got a clash here? i.e. would it make the tests fail? What's the probability of it coming up with something that's the same as a recently selected date? Just wondering if there's maybe a way to make this very unlikely to pick two dates the same by using the current times to scale things rather than just randomising it out.

Reply:
As discussed, we will just pick a new date. Adding the max number of tries will stop it retrying forever too.